packages/components/datepicker/src/components/osds-datepicker/core/controller.ts

```diff
@@ -35,6 +42,26 @@ class OdsDatepickerController {
     this.component.hasFocus = false;
     this.component.emitBlur();
   }
+
+  private getMidnightDate(date: Date) {
+    return new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()))
+  }
+
+  private validateValue(value?: Date) {
+    if (value) {
+      if (!this.component.maxDate || this.getMidnightDate(value).getTime() <= this.getMidnightDate(this.component.maxDate).getTime() ) {
```

Comment:
you should split your validation and avoid calling the same function (`getMidnightDate`) so many times, more like

```
if (!value) {
  return false;
}

const midnightDate = new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()))

if (this.component.maxDate) {
  return <test max date>
}

if (this.component.minDate) {
  return <test max date>
}

...
```